docs/source/executing-operations/native-query-planner.mdx

```diff
@@ -0,0 +1,55 @@
+---
+title: Native Query Planner
+subtitle: Run the Rust-native query planner in GraphOS Router
+minVersion: 1.49.0
+---
+
+<PreviewFeature />
+
+Learn to run the GraphOS Router with the Rust-native query planner and improve your query planning performance and scalability.
+
+## Background about query planner implementations
+
+In v1.49.0 the router introduced a [query planner](/graphos/routing/about-router#query-planning) implemented natively in Rust. This native query planner improves the overall performance and resource utilization of query planning. It exists alongside the legacy Javascript implementation. and it will eventually replace the legacy implementation.
+
+As part of the effort to ensure correctness and stability of the new query planner, starting in v1.53.0 the router can enable both the new and legacy planners and run them in parallel to compare their results. After their comparison, the router discards the native query planner's results and uses only the legacy planner to execute requests.
+
+The native query planner uses a single thread in the cold path of the router. It has a bounded queue of ten queries. If the queue is full, the router simply does not run the comparison to avoid excessive resource consumption.
+
+## Configuring query planning
+
+You can configure the `experimental_query_planner_mode` option in your `router.yaml` to set the query planner to run.
+
+The `experimental_query_planner_mode` option has the following supported modes:
+
+- `new`- enables only the new Rust-native query planner
+- `legacy` - enables only the legacy JavaScript query planner 
+- `both_best_effort` - enables both new and legacy query planners for comparison. The legacy query planner is used for execution. 
+
+<MinVersion version="2.0.0">
+
+## Optimize native query planner
+
+</MinVersion>
+
+To run the native query planner with the best performance and resource utilization, configure your router with the following options:
+
+```yaml title="router.yaml"
+supergraph:
+  generate_query_fragments: true
+
+experimental_query_planner_mode: new
+experimental_introspection_mode: new
+```
+
+- `experimental_query_planner_mode: new` enables only the Rust-native query planner.
+- `generate_query_fragments: true` enables faster throughput by reducing the size of queries sent to subgraphs.
+- `experimental_introspection_mode: new` enables Rust-native schema introspection.
+
+## Metrics for native query planner
+
+TBD
+
+## Limitations of native query planner
+
+TBD
```

Comment:
@lrlna is there any telemetry, limitations, are other topics that need to be called out on this page?